lib/algolia/analytics.rb

```diff
@@ -0,0 +1,77 @@
+module Algolia
+
+  class Analytics
+    attr_reader :ssl, :ssl_version, :headers
+
+    def initialize(client, params)
+      @client = client
+
+      @headers      = params[:headers]
+      @ssl          = params[:ssl].nil? ? true : params[:ssl]
+      @ssl_version  = params[:ssl_version].nil? ? nil : params[:ssl_version]
+    end
+
+    def get_ab_tests(params = {})
+      params = {
+          :offset => 0,
+          :limit => 10,
+      }.merge(params)
+
+      perform_request(:GET, Protocol.ab_tests_uri, params)
+    end
+
+    def get_ab_test(ab_test_id)
+      raise ArgumentError.new('ab_test_id cannot be empty') if ab_test_id.nil? || ab_test_id == ''
+
+      perform_request(:GET, Protocol.ab_tests_uri(ab_test_id))
+    end
+
+    def add_ab_test(ab_test)
+      perform_request(:POST, Protocol.ab_tests_uri, {}, ab_test.to_json)
+    end
+
+    def stop_ab_test(ab_test_id)
+      raise ArgumentError.new('ab_test_id cannot be empty') if ab_test_id.nil? || ab_test_id == ''
+
+      perform_request(:POST, Protocol.ab_tests_stop_uri(ab_test_id))
+    end
+
+    def delete_ab_test(ab_test_id)
+      raise ArgumentError.new('ab_test_id cannot be empty') if ab_test_id.nil? || ab_test_id == ''
+
+      perform_request(:DELETE, Protocol.ab_tests_uri(ab_test_id))
+    end
+
+    def wait_task(index_name, taskID, time_before_retry = WAIT_TASK_DEFAULT_TIME_BEFORE_RETRY, request_options = {})
+      @client.wait_task(index_name, taskID, time_before_retry, request_options)
+    end
+
+    private
+
+    def perform_request(method, url, params = {}, data = {})
```

Comment:
Is there a reason why we duplicate this method and don't use the client's one? We could also leverage the thread pool by using the client's.

Reply:
I dont think we can easily use the `client.perform_request` without `client.request`. And `client.request` is coupled to `@hosts`, I don't see how to override the host when calling `request()`  without modifying the client instance.

Although, I might be wrong, I'm not sure if Ruby passed a reference to the same client instance or a copy 🤔

Note that for analytics, we don't need the retry strategy nor timeouts.

Reply:
Yeah indeed it requires some adjustments on the client but it's cleaner than copying the logic IMO. It's always by "reference" kind of. You need to manually `.dup` object otherwise.

Reply:
It seems like a fairly big refactor to do. I'd rather do it later if necessary. Currently, analytics api doesn't need timouts nor retry strategy.